feat(router.js): export router injectables

PatrickJS opened the pull request:
~~`routerInjectables` should be constant `List`~~
Exporting: `RootRouter`, `RouteRegistry`, `BrowserLocation`, `Location`,  and `Pipeline`.

update: I changed the scope of this PR to only export injectables

matanlurey commented on modules/angular2/router.js:
```diff
@@ -29,7 +33,7 @@ export const routerDirectives:List = CONST_EXPR([
 ]);
 
-export var routerInjectables:List = [
+export const routerInjectables:List = CONST_EXPR([
```
This isn't a valid const list. bind() and toFactory() both are not supported.

PatrickJS committed:
feat(router.js): export router injectables

Exporting: `RootRouter`, `RouteRegistry`, `BrowserLocation`,
`Location`,  and `Pipeline`.

tbosch commented:
Travis was failing due to a flake, retrying.

vicb commented:
merged via 28ee0612cb528fc42c01e8dc413005a745b3571a
